lambda_redirect/main.tf

```diff
@@ -0,0 +1,104 @@
+/* 
+* # Lambda redirect
+* 
+* Creates a 301 redirect from a source domain to a target domain using Cloudfront and a Lambda function.
+* HTTP requests are automatically redirected to HTTPS by the CloudFront distribution.
+*/
+
+#
+# Redirector function invoked by CloudFront
+#
+resource "aws_lambda_function" "redirector" {
+  function_name = local.lambda_function_name
+  role          = aws_iam_role.redirector.arn
+  runtime       = "nodejs16.x"
+  handler       = "index.handler"
+  memory_size   = 128
+
+  filename         = data.archive_file.redirector_src.output_path
+  source_code_hash = data.archive_file.redirector_src.output_base64sha256
+
+  reserved_concurrent_executions = 1
+
+  tracing_config {
+    mode = "PassThrough"
+  }
+
+  tags = local.common_tags
+}
+
+data "archive_file" "redirector_src" {
+  type        = "zip"
+  output_path = "/tmp/redirector_src.zip"
+  source {
+    content  = <<-EOF
+      exports.handler = async (event) => {
+        return {
+          statusCode: 301,
+          headers: { 
+            Location: "${var.redirect_url}"
+          }
```

Comment:
that's a nifty idea - I've updated so that the response code and headers are now configurable.  I'm sure we'll see each other again when the `body` also gets added 😈